sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -1162,6 +1163,131 @@ case class GpuArraysOverlap(left: Expression, right: Expression)
   }
 }
 
+case class GpuMapFromArrays(left: Expression, right: Expression) extends GpuBinaryExpression {
+
+  private val mapKeyDedupPolicy = SQLConf.get.getConf(SQLConf.MAP_KEY_DEDUP_POLICY)
+
+  override def dataType: MapType = {
+    MapType(
+      keyType = left.dataType.asInstanceOf[ArrayType].elementType,
+      valueType = right.dataType.asInstanceOf[ArrayType].elementType,
+      valueContainsNull = right.dataType.asInstanceOf[ArrayType].containsNull)
+  }
+
+  def compareOffsets(lhs: ColumnVector, rhs: ColumnVector) : Boolean = {
+    val boolScalar = withResource(lhs.getListOffsetsView) { lhsOffsets =>
+      withResource(rhs.getListOffsetsView) { rhsOffsets =>
+        withResource(lhsOffsets.equalTo(rhsOffsets)) { compareOffsets =>
+          compareOffsets.all
+        }
+      }
+    }
+    withResource(boolScalar) { all =>
+      all.getBoolean
+    }
+  }
+
+  def nullSanitize(lhs: ColumnVector, rhs: ColumnVector) :
+  (ColumnVector, ColumnVector) = {
+    if(lhs.hasNulls || rhs.hasNulls) {
+      withResource(lhs.isNull) { lhsNulls =>
+        withResource(rhs.isNull) { rhsNulls =>
+          withResource(lhsNulls.or(rhsNulls)) { combinedNulls =>
+            withResource(GpuScalar.from(null, left.dataType)) { leftScalar =>
+              withResource(GpuScalar.from(null, right.dataType)) { rightScalar =>
+                val newLhs = combinedNulls.ifElse(leftScalar, lhs)
+                val newRhs = combinedNulls.ifElse(rightScalar, rhs)
+                (newLhs, newRhs)
+              }
+            }
+          }
+        }
+      }
+    }
+    else {
+      lhs.incRefCount()
+      rhs.incRefCount()
+      (lhs,rhs)
+    }
+  }
+
+  def rowContainsDuplicates(keysList: ColumnVector): Boolean = {
+    withResource(keysList.getChildColumnView(0)) { childView =>
+      withResource(keysList.dropListDuplicates) { newKeyList =>
+        withResource(newKeyList.getChildColumnView(0)) { newChildView =>
+          childView.getRowCount != newChildView.getRowCount
+        }
+      }
+    }
+  }
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+
+    require(lhs.getRowCount == rhs.getRowCount,
+      "All columns must have the same number of rows")
+
+    val (sanitizedLhsBase, sanitizedRhsBase) = nullSanitize(lhs.getBase,rhs.getBase)
+
+    val nullKeysCount = withResource(sanitizedLhsBase.getChildColumnView(0)) { childView =>
+      childView.getNullCount
+    }
+
+    require(nullKeysCount == 0,
+      "[NULL_MAP_KEY] Cannot use null as map key")
+
+    if(mapKeyDedupPolicy == "EXCEPTION") {
+      val containsDuplicates = rowContainsDuplicates(sanitizedLhsBase)
+      require(!containsDuplicates,
+        "[DUPLICATED_MAP_KEY] Duplicate map key was found")
+    }
+
+    require(compareOffsets(sanitizedLhsBase, sanitizedRhsBase),
```

Comment:
nit: it looks like null checks for the keys in Spark on the CPU happens after the length check.

Reply:
Done!